### DISCUSSION TITLE
Enabling IDN support by default on Windows (possibly followed by macOS)

### DISCUSSION THREAD

**Discussion by vszakats** [curl/curl#13565] [Ideas]
IDN support is currently disabled by default for Windows builds with both CMake and autotools.

The reason for this is Windows XP not supporting the IDN library out of the box. It requires a Microsoft package to be installed manually. Microsoft pulled this package from the internet years ago, so finding / installing it isn't trivial. XP is the oldest version we support since 960d601481eed46d7b46391380c4869bcf0d9f19 #12225 (2023-10-28).

Would it be time to change these build options to enable IDN and thus potentially break the binary on Windows XP systems by default? It would remain possible to target XP by disabling the build option manually.

FWIW curl-for-win is already building with IDN support. Microsoft enables it too in their builds. vcpkg has it as an option, but perhaps not enabled by default, based on (mis?)reading the [package definition](https://github.com/microsoft/vcpkg/blob/master/ports/curl/portfile.cmake).

If done, we might follow up doing the same for macOS (but that first needs more testing with AppleIDN enabled, as well as a `./configure` option for the feature.)

TL;DR:

- is it desired to have IDN support enabled by default if the OS has built-in support it?
- are we okay to change build defaults to enable it by default? (and potentially breaking Windows XP by default)

Any thoughts on these?

**Comment by jay** [comment]
Seems reasonable for Windows. We already do it for the winbuild build method

https://github.com/curl/curl/blob/de7b3e89218467159a7af72d58cea8425946e97d/winbuild/Makefile.vc#L73-L79

**Comment by bagder** [comment]
I'm totally fine with doing this.
